pkg/utils/utils.go

```diff
@@ -30,4 +33,61 @@ func NullFloatToNullableTime(ts null.Float) *time.Time {
 	return &timestamp
 }
 
+func FlattenNestedFieldsToObj(field map[string]interface{}) map[string]interface{} {
+	// from "span.attributes.sampler@type": "test"
+	// to map[span:map[attributes:map[sampler@type:test]]]
+	result := make(map[string]interface{})
+
+	for key, value := range field {
+		keys := strings.Split(key, ".")
+		current := result
+
+		for i := 0; i < len(keys)-1; i++ {
+			if _, exists := current[keys[i]]; !exists {
+				current[keys[i]] = make(map[string]interface{})
+			}
+			current = current[keys[i]].(map[string]interface{})
+		}
+
+		current[keys[len(keys)-1]] = value
+	}
+
+	return result
+}
+
+func TimeFieldToMilliseconds(date interface{}) (int64, error) {
+	layout := "2006-01-02T15:04:05.000000Z"
+	var timestamp *int64
+	switch timeField := date.(type) {
+	case string:
+		t, err := time.Parse(layout, timeField)
+		if err != nil {
+			return 0, err
+		}
+		nano := t.UnixNano() / 1e6
+		timestamp = &nano
+	case int64:
+		timestamp = &timeField
+	default:
+		return 0, errors.New("unrecognized time format")
+	}
+	return *timestamp, nil
+}
+func SpanHasError(spanEvents []interface{}) bool {
+	for _, event := range spanEvents {
+		if eventMap, ok := event.(map[string]interface{}); ok {
```

Comment:
also obnoxious style nit: You could do
```
eventMap, ok := event.(map[string]interface{})
if !ok {
   log.DefaultLogger.Debug("span event is not a map")
   continue
}
attributes := eventMap["attributes"]
...
```
(The idea behind the style nit is returning/continuing early and reducing indents)